### PR TITLE
added options page & configurable separator

### PR DESCRIPTION
**Pull request by Azlond**
added options to manifest.json
added options.html + options.js. options.html contains a form to change
the default ("|") separator, options.js saves changes to the addons
storage.
Adjusted titleUrlPage.js to work with the customized separator, which
has to be received via promise.

I've left the version number unchanged, and haven't paid attention to locales - only language supported in this PR is English. The options page is kept very simple and could probably use adjustments.
I had to rearrange titleUrlPage.js a bit to work with the storage-promise. Leaving it as title=getTitle() would have always returned undefined, due to the return statement executing faster than the promise.

Hope this is similiar to what you had in mind with #5.

### CODE REVIEW

File: addon/options/options.html [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<!DOCTYPE html>
+<html>
+<head>
+	<meta charset="utf-8">
+</head>
+<body>
+	<form id="form">
+		<input type="text" name="separatorField" id="separatorField">
+		</textarea>
```
Comment (cloutierjo) [inline addon/options/options.html:9]: I suppose the textarea isn't necessary

File: addon/titleUrlPage.js [hunk 2]
```diff
@@ -3,20 +3,20 @@ browser.runtime.onMessage.addListener(updateUrl)
...
-    } else {
-        return document.title;
-    }
+	function updateTitle(result) {
+		let separator = "|";
+		if (result.separator) {
+			separator = result.separator;
+		}
+		document.title = document.title + " " + separator + " " + url;
+	}
```
Comment (cloutierjo) [inline addon/titleUrlPage.js]: I understand why the change is required by using the promise in that way. I would probably have used an options object that would be initialized  at page load and updated on storage change. This is to keep the updateUrl work the most simple as possible because some website updates their title quite often and we need to run this function on every title change.

**Review by cloutierjo — COMMENTED** [review]
Thanks for the PR most of it looks good, I'll do some testing about the promise usage to be certain that it doesn't cause a problem, but if it's working well I'll merge it like this.

File: addon/options/options.html (hunk 1 above)
Comment (Azlond) [reply addon/options/options.html:9]: Yeah, it's not supposed to be there. I had thought about working with a textarea, but decided on the input/form instead. I seem to have missed that closing textarea tag though.

File: addon/titleUrlPage.js (hunk 2 above)
Comment (Azlond) [reply addon/titleUrlPage.js]: Ah, I hadn't thought about that. Makes sense.

**Comment by Azlond** [comment]
Another feature I've not implented here: When you first install the addon, all tab titles get updated. It might be nice to also update tab titles when saving a new separator.

**Comment by cloutierjo** [comment]
Afayk, they are not all updated on install, but each time a tab is activated, so that would work in case of settings change as the settings are in another tabs (might have to check with multiple windows though).

**Comment by Azlond** [comment]
I just tested the current version (0.2) with multiple windows. As soon as the addon is enabled, all tab titles (across all windows) are updated with the URL.

**Comment by cloutierjo** [comment]
Ok, I've done more tests and there are problems updating the URL on some website, at least SPA kind like github. The URL is appended more than once or not at all in some case.

I also found that I had a bug with the update on tab change feature, I'll fix that in a few minutes on master. Still even with that fix, the setting is not taken into account without a page reload.

So in the current state, I can't merge. The SPA site case would need to be fixed before.

**Comment by Azlond** [comment]
I'll try to reproduce the bug and see if I can fix it.

**Comment by cloutierjo** [comment]
It seam to work well now. I'll release a beta version to tests it for a few day, thank for your contribution.

**Comment by schmittlauch** [comment]
Maybe one more UX thing: Currently there's no confirmation of saving the separation character(s).
I'd suggest deactivating the "Save" button if there are no changes to save, e.g. when no changes have been made since the last save.
